tests/test_core.py

```diff
@@ -302,6 +302,27 @@ def test_update(self):
         with self.assertRaises(KeyError):
             d.update(C=2.5)
 
+    def test_clear(self):
+        """Test clear method to reset keys to default."""
+
+        #test clear with no default set
+        d = relentless.core.FixedKeyDict(keys=('A','B'))
```

Comment:
For both of these, can you check the values of `d['A']` and `d['B']` before you call the update method? Just to make sure the constructor is working right.